Composer/packages/client/src/components/ManageService/ManageService.tsx

```diff
@@ -0,0 +1,702 @@
+    },
+    { key: 'handoff', text: formatMessage('Handoff to admin') },
+  ];
+
+  const fetchLocations = async (subscriptionId) => {
+    if (token) {
+      const tokenCredentials = new TokenCredentials(token);
+      const subscriptionClient = new SubscriptionClient(tokenCredentials);
+      const locations = await subscriptionClient.subscriptions.listLocations(subscriptionId);
+      setLocationList(
+        locations.map((location) => {
+          return { key: location.name || '', text: location.displayName || '' };
+        })
+      );
+    }
+  };
+
+  /* Copied from Azure Publishing extension */
+  const getSubscriptions = async (token: string): Promise<Array<Subscription>> => {
+    const tokenCredentials = new TokenCredentials(token);
+    try {
+      const subscriptionClient = new SubscriptionClient(tokenCredentials);
+      const subscriptionsResult = await subscriptionClient.subscriptions.list();
+      // eslint-disable-next-line no-underscore-dangle
+      return sortBy(subscriptionsResult._response.parsedBody, ['displayName']);
+    } catch (err) {
+      setApplicationLevelError(err);
+      return [];
+    }
+  };
+
+  const hasAuth = async () => {
+    let newtoken = '';
+    if (userShouldProvideTokens()) {
+      if (isShowAuthDialog(false)) {
+        setShowAuthDialog(true);
+      }
+      newtoken = getTokenFromCache('accessToken');
+    } else {
+      newtoken = await AuthClient.getAccessToken(armScopes);
+    }
+
+    setToken(newtoken);
+
+    if (newtoken) {
+      // reset the list
+      setAvailableSubscriptions([]);
+
+      // fetch list of available subscriptions
+      setAvailableSubscriptions(await getSubscriptions(newtoken));
+
+      // go on to the next page post-auth
+      setCurrentPage(2);
+    }
+  };
+
+  useEffect(() => {
+    // reset the ui
+    setSubscription('');
+    setKeys([]);
+    setCurrentPage(1);
+  }, [props.hidden]);
+
+  const handleRegionOnChange = (e, value: IDropdownOption | undefined) => {
+    if (value != null) {
+      setRegion(value.key as string);
+    } else {
+      setRegion('');
+    }
+  };
+
+  const fetchKeys = async (cognitiveServicesManagementClient, accounts) => {
+    const keyList: KeyRec[] = [];
+    for (const account in accounts) {
+      const resourceGroup = accounts[account].id?.replace(/.*?\/resourceGroups\/(.*?)\/.*/, '$1');
+      const name = accounts[account].name;
+      if (resourceGroup && name) {
+        const keys = await cognitiveServicesManagementClient.accounts.listKeys(resourceGroup, name);
+        if (keys?.key1) {
+          keyList.push({
+            name: name,
+            resourceGroup: resourceGroup,
+            region: accounts[account].location || '',
+            key: keys?.key1 || '',
+          });
+        }
+      }
+    }
+    return keyList;
+  };
+
+  const fetchAccounts = async (subscriptionId) => {
+    if (token) {
+      setLoading(formatMessage('Loading keys...'));
+      setNoKeys(false);
```

Comment:
before you fetch the keys, no keys should be ```true``` meaning no keys right?

Reply:
the noKeys flag should only be true if the query has returned no keys.